pkg/controller/schedule_controller.go

```diff
@@ -89,6 +89,11 @@ func (c *scheduleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, errors.Wrapf(err, "error getting schedule %s", req.String())
 	}
 
+	if schedule.Spec.Paused {
+		log.Info("the schedule is paused, skip")
```

Comment:
To me, I think this checking can be put into `PeriodicalEnqueueSource` fitlerFuncs.
If the schedule is in pending state, no updating should be reasonable.

I also put the generated backup state checking into fitler functions, maybe it should be put into reconcile function, because not handling schedule that has backup in progress may be a surprise for user.